### ISSUE TITLE
Incompatibility with Ruby 3

### ISSUE THREAD

**Issue by ioquatix** [guard/guard#968]
https://github.com/guard/guard/blob/24e865c77bcce9058d28a393d0799614902a60e1/lib/guard/plugin_util.rb#L65

This line was previously compatible with Ruby 2.x when plugins had the following format:

```ruby
class MyPlugin
  def initialize(**options)
    # ...
  end
end
```

However in Ruby this this usage becomes an error:

```
11:39:01 - INFO - Bye bye...
11:39:01 - ERROR - Invalid Guardfile, original error is:
> [#]
> [#] Failed to call Guard::Falcon.new(options): wrong number of arguments (given 1, expected 0),
> [#] backtrace:
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/plugin_util.rb:64:in `rescue in initialize_plugin'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/plugin_util.rb:61:in `initialize_plugin'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/internals/plugins.rb:26:in `add'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/dsl.rb:186:in `block in guard'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/dsl.rb:183:in `each'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/dsl.rb:183:in `guard'
> [#] 	(dsl)> ./Guardfile:3:in `block in evaluate'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/dsl.rb:142:in `group'
> [#] 	(dsl)> ./Guardfile:2:in `evaluate'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/dsl.rb:378:in `instance_eval'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/dsl.rb:378:in `evaluate'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/guardfile/evaluator.rb:95:in `evaluate'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard.rb:134:in `_evaluate'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard.rb:49:in `setup'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/commander.rb:32:in `start'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/cli/environments/valid.rb:16:in `start_guard'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/cli.rb:122:in `start'
> [#] 	(dsl)> $GEM_PATH[0]/gems/thor-1.1.0/lib/thor/command.rb:27:in `run'
> [#] 	(dsl)> $GEM_PATH[0]/gems/thor-1.1.0/lib/thor/invocation.rb:127:in `invoke_command'
> [#] 	(dsl)> $GEM_PATH[0]/gems/thor-1.1.0/lib/thor.rb:392:in `dispatch'
> [#] 	(dsl)> $GEM_PATH[0]/gems/thor-1.1.0/lib/thor/base.rb:485:in `start'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/aruba_adapter.rb:32:in `execute'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/lib/guard/aruba_adapter.rb:19:in `execute!'
> [#] 	(dsl)> $GEM_PATH[0]/gems/guard-2.16.2/bin/_guard-core:11:in `<main>'
```

It's caused by this line: https://github.com/guard/guard/blob/master/lib/guard/plugin_util.rb#L65

@mame what is the best way to fix this? I don't think this is a use case of `ruby2_keywords` right? It's kind of the inverse issue.

It seems like we should probably try to sniff whether it should be a positional argument or keyword arguments.

cc @pharmac1st

**Comment by rymai** [comment]
Would calling `klass.new(**options)` solve this? 🤔

**Comment by ioquatix** [comment]
Yes but it potentially breaks some existing plugins.

**Comment by rymai** [comment]
> Yes but it potentially breaks some existing plugins.

I think it should work fine for `def initialize(options = {})` and `def initialize(**options)`:

```ruby
[1] pry(main)> def foo(opts = {})
[1] pry(main)*   puts opts
[1] pry(main)* end  
=> :foo
[2] pry(main)> foo({a: 1})
{:a=>1}
=> nil
[3] pry(main)> foo(**{a: 1})
{:a=>1}
=> nil
[4] pry(main)> def bar(**opts)
[4] pry(main)*   puts opts
[4] pry(main)* end  
=> :bar
[5] pry(main)> bar({a: 1})
ArgumentError: wrong number of arguments (given 1, expected 0)
from (pry):6:in `bar'
[6] pry(main)> bar(**{a: 1})
{:a=>1}
=> nil
```

**Comment by rymai** [comment]
I've released Guard 2.17.0 with this change and I created a discussion for it here: https://github.com/guard/guard/discussions/972

**Comment by ioquatix** [comment]
Thanks for sorting this out!

cc @pharm1cist
